src/Components/Shifting/ShiftingBoard.tsx

```diff
@@ -274,66 +271,32 @@ export default function ShiftingBoard({
     collect: (monitor) => ({ isOver: !!monitor.isOver() }),
   }));
 
-  const fetchData = () => {
-    setIsLoading((loading) => reduceLoading("BOARD", loading));
-    dispatch(
-      listShiftRequests(formatFilter({ ...filterProp, status: board }), board)
-    ).then((res: any) => {
-      if (res && res.data) {
-        setData(res.data.results);
-        setTotalCount(res.data.count);
+  const { data, refetch } = useQuery(routes.listShiftRequests, {
+    query: formatFilter({
+      ...filterProp,
+      status: board,
+      offset: offset,
+    }),
+    onResponse: ({ res, data }) => {
+      if (res?.ok && data) {
         setCurrentPage(1);
       }
       setIsLoading((loading) => reduceLoading("COMPLETE", loading));
-    });
-  };
-
-  useEffect(() => {
-    fetchData();
-  }, [
-    board,
-    dispatch,
-    filterProp.facility,
-    filterProp.origin_facility,
-    filterProp.shifting_approving_facility,
-    filterProp.assigned_facility,
-    filterProp.emergency,
-    filterProp.is_up_shift,
-    filterProp.patient_name,
-    filterProp.created_date_before,
-    filterProp.created_date_after,
-    filterProp.modified_date_before,
-    filterProp.modified_date_after,
-    filterProp.patient_phone_number,
-    filterProp.ordering,
-    filterProp.is_kasp,
-    filterProp.assigned_to,
-    filterProp.disease_status,
-    filterProp.is_antenatal,
-    filterProp.breathlessness_level,
-  ]);
+    },
+  });
 
   const handlePagination = (page: number, limit: number) => {
     const offset = (page - 1) * limit;
     setCurrentPage(page);
+    setOffSet(offset);
     setIsLoading((loading) => reduceLoading("MORE", loading));
-    dispatch(
-      listShiftRequests(
-        formatFilter({ ...filterProp, status: board, offset: offset }),
-        board
-      )
-    ).then((res: any) => {
-      if (res && res.data) {
-        setData((data) => [...data, ...res.data.results]);
-        setTotalCount(res.data.count);
-      }
-      setIsLoading((loading) => reduceLoading("COMPLETE", loading));
-    });
+    refetch();
```

Comment:
You don't need to call `refetch` if the query params changes. Doing so will cause refetch twice, since `useQuery` will automatically perform refetch too whenever qParams changes.